### PR TITLE
Connection changes ignite

### PR DESCRIPTION
**Pull request by MilesHolland**
# Description

Makes some changes to the workspace connection entity that are needed for ignite, including:
 - making various fields that were previous part of metadata top-level. These are still pushed into metadata when converted to a a rest object. Fields include api_version, api_type, and kind.
 - remove metadata as a field - users can use tags instead.
 - Unrelated tiny fix: Add none protection around arm_id check. This fixes a bug where hub deletion would fail if delete_dependent_resources was set to true in the operation call.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by joshharrin — APPROVED** [review]
thanks for the help with Hub deletion fix. other changes lgtm

File: sdk/ml/azure-ai-ml/azure/ai/ml/constants/_common.py [hunk 1]
```diff
@@ -182,6 +182,9 @@
     "Spark job will only install the packages defined in the Conda configuration. It "
     "will not create a docker container using the image defined in the environment."
 )
+API_VERSION_KEY = "ApiVersion"
```
Comment (ninghu) [inline sdk/ml/azure-ai-ml/azure/ai/ml/constants/_common.py:185]: The names for those constants are too generic. Perhaps consider adding a "CONNECTION_" prefix for clarity.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/connections/workspace_connection.py [hunk 2]
```diff
@@ -125,29 +145,46 @@ def credentials(
...
 
     @property
-    def metadata(self) -> Dict[str, Any]:
-        """Metadata for workspace connection.
+    def api_version(self) -> str:
+        """The API version of the workspace connection.
 
-        :return: Metadata for workspace connection.
-        :rtype: Dict[str, Any]
+        :return: the API version of the workspace connection.
```
Comment (ninghu) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/connections/workspace_connection.py:166]: the -> The

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/connections/workspace_connection.py [hunk 3]
```diff
@@ -125,29 +145,46 @@ def credentials(
...
+        :rtype: str
         """
-        return self._metadata
+        return self._api_version
+
+    @property
+    def api_type(self) -> str:
+        """The API type of the workspace connection.
+
+        :return: the API type of the workspace connection.
```
Comment (ninghu) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/connections/workspace_connection.py:175]: same here

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/connections/workspace_connection_subtypes.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    ApiKeyConfiguration,
+)
+from azure.ai.ml.entities._util import load_from_dict
+from .workspace_connection import WorkspaceConnection
+
+
+# Dev notes: Any new classes require modifying the elif chains in the following functions in the
+# WorkspaceConnection parent class: _from_rest_object, _get_entity_class_from_type, _get_schema_class_from_type
+@experimental
+class OpenAIWorkspaceConnection(WorkspaceConnection):
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/connections/workspace_connection_subtypes.py:37]: Let's call this AzureOpenAIWorkspaceConnection. Native OpenAI doesn't care about `api_version` and `api_type` for example.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_load_functions.py [hunk 5]
```diff
@@ -736,7 +746,62 @@ def load_workspace_connection(
...
+
+            def build_error(message, _):
+                from azure.ai.ml.entities._util import decorate_validation_error
+
+                return ValidationError(
+                    message=decorate_validation_error(
+                        schema=schema.__class__,
+                        pretty_error=message,
+                        additional_message=additional_message,
+                    ),
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_load_functions.py:800]: Can these duplicate logic be removed now?

**Review by luigiw — CHANGES_REQUESTED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_load_functions.py (hunk 5 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_load_functions.py:800]: Yes, already removed. The commit just didn't go through.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/connections/workspace_connection_subtypes.py (hunk 4 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/connections/workspace_connection_subtypes.py]: Changed

**Pull request merged**